### ISSUE TITLE
Run a smoke test on Docker/Jenkins using a basic pytest on windows

### ISSUE THREAD

**Issue by dsfai2020** [dsfai2020/Task-and-Nutrition-Manager#43]
This process is to take place on Windows 10 using Docker.  While also having Jenkins installed on a container that not only runs, but supports pytest from within docker as well.

### ON WINDOWS 10

- [x]  Smoke test a python app that runs through powershell
- [ ] Get a python docker container file up and running on windows 

### ON JENKINS/DOCKER
- [x] make sure that it activates a venv and installs the packages from the python env of the repo from the terminal
- [ ] make sure that properly runs a basic test
- [ ] make sure that it then passes the test
- [ ] ensure that the script for jenkins pulls the project in from github to its jenkins directory (which is hosted within Docker)

- [ ] Once that is done, connect multiple containers (python and Jenkins) so that python is supported within Docker.  

Failure in any of these areas results in a non completion.

**Comment by dsfai2020** [comment]
Ran into a couple issues.

- [ ] - then Jenkins container from the docker recommended link on its page doesn't include Python.  This means that python scripts like pytest or venv don't exist at all in the terminal when executing scripts on Jenkins.  But it seems to work just fine if you install it locally on Linux.  My guess is that the docker container needs to have Python in it or be connected to another container via mount so that the Jenkins container can have access to it.

Apt-get won't even work at the moment.  Tried configuring it through the docker terminal and ran into a dead end.  Permission was denied.  Sudo as well.

As it stands, The docker image works.  Jenkins runs on the local ports it makes.  It supports Debian but it just doesn't seem to come pre configured with python.  

- [ ] Check how to mount another container to it that DOES have python that way they are isolated.

- YAML config most likely needed

**Comment by dsfai2020** [comment]
https://docs.docker.com/get-started/07_multi_container/

**Comment by dsfai2020** [comment]
Python lead for docker here.  (You'll need it for pytest in Jenkins and eventually you'll need it for dev servers on the backend.

https://docs.docker.com/language/python/build-images/

**Comment by dsfai2020** [comment]
powershell needed a local python interpreter so I downloaded one from the store windows store to simplify the process on Windows.

![image](https://github.com/dsfai2020/shopsite/assets/60169737/e348cbfd-94ff-4aaf-88a7-2d0a28f7f74f)

![image](https://github.com/dsfai2020/shopsite/assets/60169737/5fe6973b-1d28-411d-ba83-575336f9a6ec)

**Comment by dsfai2020** [comment]
> powershell needed a local python interpreter so I downloaded one from the store windows store to simplify the process on Windows.
> 
> ![image](https://user-images.githubusercontent.com/60169737/256762045-5fe6973b-1d28-411d-ba83-575336f9a6ec.png)
> 
> ![image](https://user-images.githubusercontent.com/60169737/256761944-e348cbfd-94ff-4aaf-88a7-2d0a28f7f74f.png)
> 
> Ran into a new issue where python isn't being detected on the PATH for Windows powershell OR command prompt.

Resolved with Information from

https://stackoverflow.com/questions/58754860/cmd-opens-windows-store-when-i-type-python

I had to go into the env settings and move the directory that houses the windows apps (since it was from the store) to the top of the list.  I did this for both local and global.  Then I went into the Alias page and disabled and re-enabled the windowed mode and regular mode pythons.  This fixed the problem.

**Comment by dsfai2020** [comment]
Now that python is working run these commands to get the venv up and running.

`python -m venv nameofenv`
`nameofenv\Scripts\activate`

**Comment by dsfai2020** [comment]
Ran into an issue where docker wouldn't build on powershell because the Docker Daemon wasn't running.
To resolve this I'll attempt to run powershell as Admin.

**Comment by dsfai2020** [comment]
This concludes https://docs.docker.com/language/python/build-images/

Next up is:

https://docs.docker.com/language/python/run-containers/
